realm/src/main/java/io/realm/Realm.java

```diff
@@ -1403,6 +1694,73 @@ public void executeTransaction(Transaction transaction) {
     }
 
     /**
+     * Similar to {@link #executeTransaction(Transaction)} but runs asynchronously from a worker thread
+     * @param transaction {@link io.realm.Realm.Transaction} to execute.
+     * @param callback optional, to receive the result of this query
+     * @return A {@link io.realm.Realm.Request} representing a cancellable task
+     */
+    public Request executeTransaction(final Transaction transaction, final Transaction.Callback callback) {
```

Comment:
I prefer `executeTransactionAsync` to overloaded `executeTransaction`.
Overloaded sync and async methods are a bit confusing to users.

I think there is a big difference from the case of Retrofit.
In Retrofit case, overloaded methods are defined intentionally by users.
Of course they know these methods are sync and async.

But in this case, API is given, and they can't guess `realm.executeTransaction(new Realm.Transaction() { ... }, null);` is sync or async.

In addition to that, Retrofit quit to use that overloaded style in 2.0(the reason is different from this case, though).